californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -976,13 +978,7 @@ private void receiveEmptyMessage(final EmptyMessage message) {
 
 							@Override
 							public void runStriped() {
-								int mid;
-								if (exchange.getOrigin() == Origin.LOCAL) {
-									mid = exchange.getCurrentRequest().getMID();
-								} else {
-									mid = exchange.getCurrentResponse().getMID();
-								}
-								if (message.getMID() == mid) {
+								if (exchange.checkMID(message.getMID())) {
```

Comment:
Is it not to the matcher to return exchange only if it match this message ?
(I must to confess, I do not really understand why we need to do this check ?)

Reply:
:-). That matching may change over the time, therefore I revalidate it within the stripped execution.

Reply:
Ok we do the same "kind" of check when receiving response too.

I think this maybe deserve some comment in the code to explain that because it's important to not change this and we don't have tests which could ensure that this is not modified by mistake.

Reply:
OK, I added two comments.